### PR TITLE
Prevent exception upon HEAD requests to the download view of a file.

### PR DESCRIPTION
**Pull request by mbaechtold**
⚠️This also drops support for Plone 4.2. ⚠️

### CODE REVIEW

**Comment by mbaechtold** [comment]
Thanks to @buchi for advising how to solve this.

File: ftw/file/tests/test_download.py [hunk 1]
```diff
@@ -94,6 +94,26 @@ def test_inline_download(self, browser):
...
+        browser.login().open(self.context, view='@@download', method='HEAD')
+
+        # We get a 302 but the testbrowser follows the redirect.
+        self.assertEqual(
+            'http://nohost/plone/f1/@@download/file/file.doc',
+            browser.url
+        )
+
+        # Let's open the url we got from the redirect.
+        browser.login().open(browser.url, method='HEAD')
```
Comment (jone) [inline ftw/file/tests/test_download.py]: 🤔 das ist eigentlich nicht notwendig - das ist bereits passiert (wegen automatischem Redirect).
Das Login ist auch doppelt.

Comment (mbaechtold) [reply ftw/file/tests/test_download.py]: @jone, I have reworded the comment in the test to make my intentions more clear. What do you think?

File: ftw/file/tests/test_download.py [hunk 2]
```diff
@@ -94,6 +94,30 @@ def test_inline_download(self, browser):
...
                           browser.headers['content-disposition'])
 
+    @browsing
+    def test_head_request(self, browser):
+        browser.default_driver = LIB_TRAVERSAL
+
+        browser.login()
+
+        # Accessing "@@download" on the file would normally result in a 302,
+        # telling us to got to "@@download/file/file.doc" instead.
```
Comment (jone) [inline ftw/file/tests/test_download.py]: Typo `got to` => `go to`

File: ftw/file/tests/test_download.py [hunk 3]
```diff
@@ -94,6 +94,30 @@ def test_inline_download(self, browser):
...
+        # But since the testbrowser follows the redirects, we get a 200 instead.
+        browser.open(self.context, view='@@download', method='HEAD')
+        self.assertEqual(200, browser.status_code)
+        self.assertEqual(
+            'http://nohost/plone/f1/@@download/file/file.doc',
+            browser.url
+        )
+
+        # Open the correct url used to access the file.
+        browser.login().open(self.context, view='@@download/file/file.doc', method='HEAD')
```
Comment (jone) [inline ftw/file/tests/test_download.py]: I'd leave the `login()` since that has already happened.

File: ftw/file/tests/test_download.py (hunk 1 above)
Comment (jone) [reply ftw/file/tests/test_download.py]: Fair 👍 Added comments.

File: ftw/file/tests/test_download.py (hunk 3 above)
Comment (mbaechtold) [reply ftw/file/tests/test_download.py]: I have removed it since the login happens earlier.

File: ftw/file/tests/test_download.py (hunk 2 above)
Comment (mbaechtold) [reply ftw/file/tests/test_download.py]: I have fixed the typo.

**Comment by mbaechtold** [comment]
@jone, could you please have another look?

**Pull request merged**